File: selling-used-items-app-backend/selling-used-items-app-backend/Model/Purchase.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using selling_used_items_app_backend.Enum;

namespace selling_used_items_app_backend.Model
{
    public class Purchase
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int id { get; set; }

        [ForeignKey("User")]
        public int userId { get; set; }

        [ForeignKey("Advertisement")]
        public int advertisementId { get; set; }
    }
}

File: selling-used-items-app-frontend/src/app/model/purchase.ts
export class Purchase {
    id: number = 0;
    userId: number = 0;
    advertisementId: number = 0;
  
    constructor(
      id: number = 0,
      userId: number = 0,
      advertisementId: number = 0
    ) {
      this.id = id;
      this.userId = userId;
      this.advertisementId = advertisementId;
    }
  }
  